python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-        """
-
-        new_metadata = TableMetadata(
-            name=table_name,
-            description=metadata["tableDescription"],
-            column_details=metadata["columns"],
-            retention_period_in_days=retention_period,
-            source_dataset_name=metadata.get("sourceDatasetName", ""),
-            where_to_access_dataset=metadata.get("sourceDatasetLocation", ""),
-            data_sensitivity_level=SecurityClassification[
-                metadata.get("securityClassification", "OFFICIAL")
-            ],
-            tags=metadata.get("tags", []),
-        )
-
-        return new_metadata
-
-
-# jscpd:ignore-start
-@dataclass
-class DatabaseMetadata:
-    """
-    For source system databases - currently matches DataProductMetadata
-    but will need to be revisted and refined potentially
-    """
-
-    name: str
-    description: str
-    version: str
-    owner: str
-    owner_display_name: str
-    maintainer: str | None
-    maintainer_display_name: str | None
-    email: str
-    retention_period_in_days: int
-    domain: str
-    subdomain: str | None
-    dpia_required: bool
-    dpia_location: str | None
-    last_updated: datetime
-    creation_date: datetime
-    s3_location: str | None
-    status: DataProductStatus = DataProductStatus.DRAFT
-    tags: list[str] = field(default_factory=list)
-
-
-# jscpd:ignore-end
-
-
-@dataclass
-class ChartMetadata:
-    name: str
-    description: str
-    external_url: str
-    tags: list[str] = field(default_factory=list)
+class Column(BaseModel):
+    """
+    A column definition in a table
+    """
+
+    name: str = Field(
+        pattern=r"^[\w.]+$",
+        description="The name of a column as it appears in the table.",
+        json_schema_extra={"pattern": r"^[\w.]+$"},
+    )
+    display_name: str = Field(description="A user-friendly version of the name")
+    type: str = Field(
+        description="The data type of the column as it appears in the table",
+    )
+    description: str = Field(description="A description of the column")
+    nullable: bool = Field(description="Whether the field is nullable or not")
+    is_primary_key: bool = Field(
+        description="Whether the field is part of the primary key"
+    )
+    foreign_keys: list[ColumnRef] | None = Field(
+        description="References to columns in other tables"
+    )
+
+
+class OwnerRef(BaseModel):
+    """
+    A reference to a named individual that performs some kind of governance
+    """
+
+    display_name: str = Field(
+        description="The full name of the user as it should be displayed"
+    )
+    email: str = Field("Contact email for the user")
+    id: str = Field("Unique identifier for the user")
+
+
+# Needs refining
+class Governance(BaseModel):
+    """
+    Governance model for an entity or domain
+    """
+
+    data_owner: OwnerRef = Field(description="")
+    data_stewards: list[OwnerRef] = Field(description="")
+
+
+class DomainRef(BaseModel):
```

Comment:
Question: do you think it's worth dropping `DomainRef` and `TagRef` and just using `EntityRef`

`OwnerRef` I deliberately made separate as there was some different fields, but in this case I'm not sure it's adding anything other than a bit of type safety.

Reply:
Wondering if there might be additional properties we add to domain or tags, suppose both might benefit with having descriptions? And i think Datahub does allow descriptions for tags and domains